### DISCUSSION TITLE
Bookmark Toogle

### DISCUSSION THREAD

**Discussion by mohbusaidy** [neuefische/questions#38] [Web] [answered]
Hi leude, 

ich versuche vergäblich das sich mein Herz Icon ändert wenn es sich beim favorite feld hinzufügt. Ich habe versucht 2 verschiedene icons zu nutzen. Un habe eine "turn operator" funktion geschrieben, jetzt hat sich meine ganze Seite aufgehangen. 

Der [link](https://github.com/mohbusaidy/Capstone-Project/tree/favorites) zum Branch.

**Comment by ThomasF85** [comment, marked as answer]
Hi,

ich kann bei dir mehrere Fehler beim Kompilieren feststellen. Diese solltest du zuerst beheben, damit die Applikation überhaupt startet:

1. In den Dateien BookmarkIcon.jsx und NoBookmarkIcon.jsx sind deine Komponenten syntaktisch falsch definiert und exportiert. Du kannst deine Komponenten z.B. so exportieren:

```
const BookmarkIcon = () => {
  ...
};

export default BookmarkIcon;
```

2. In der Product.jsx benutzt du die Variable **bookmarked**. Die Variable ist aber nicht definiert - ich vermute das war ursprünglich dein Problem. Hier musst du dir überlegen woher die Information in der Variable kommt (sollte sie z.B. als prop übergeben werden?). Du kannst die Zeile hier zunächst auskommentieren, damit deine Applikation wieder läuft um dann das eigentliche Problem zu lösen.

**Reply by mohbusaidy** [reply]
Meine applikation läuft wieder hab eben gerade auch gepusht. Wollte ganz gerne wissen wie der lösungsansatz sei für die toggle funktion.
